### PR TITLE
detect-secrets: 0.12.4 -> 0.14.3

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.14.3

Tests are now requiring [pyahocorasick](https://github.com/WojciechMula/pyahocorasick).

Changelog: https://github.com/Yelp/detect-secrets/blob/master/CHANGELOG.md

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pyahocorasick/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyahocorasick/default.nix:3]:
```suggestion
, fetchFromGitHub
```

File: pkgs/development/tools/detect-secrets/default.nix [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
 
-  propagatedBuildInputs = [ pyyaml ]
+  propagatedBuildInputs = [ pyyaml requests ]
     ++ lib.optionals isPy27 [ configparser enum34 future functools32 ];
 
-  checkInputs = [ mock pytest unidiff ];
+  checkInputs = [
+    mock
+    pyahocorasick
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/tools/detect-secrets/default.nix:36]: Can we use pythonCheckPhase?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@fabaff please fix the eval error.

File: pkgs/development/tools/detect-secrets/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/tools/detect-secrets/default.nix]: Yes, seems possible.

File: pkgs/development/python-modules/pyahocorasick/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyahocorasick/default.nix]: Updated, thanks.

**Comment by fabaff** [comment]
> @fabaff please fix the eval error.

@SuperSandro2000, done.

File: pkgs/development/python-modules/pyahocorasick/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "pyahocorasick";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "WojciechMula";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyahocorasick/default.nix:14]:
```suggestion
    rev = version;
```

File: pkgs/development/tools/detect-secrets/default.nix [hunk 4]
```diff
@@ -1,29 +1,58 @@
...
+
+  pythonImportsCheck = [ "detect_secrets" ];
+
+  # # deselect tests which require git setup
+  # checkPhase = ''
+  #   PYTHONPATH=$PWD:$PYTHONPATH pytest \
+  #     --deselect tests/main_test.py: \
+  #     --deselect tests/pre_commit_hook_test.py:: \
+  #     --deselect tests/core/baseline_test.py::
+  # '';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/detect-secrets/default.nix:55]:
```suggestion
```

File: pkgs/development/tools/detect-secrets/default.nix [hunk 5]
```diff
@@ -1,29 +1,58 @@
...
-    repo = "detect-secrets";
+    repo = pname;
     rev = "v${version}";
-    sha256 = "01y5xd0irxxib4wnf5834gwa7ibb81h5y4dl8b26gyzgvm5zfpk1";
+    sha256 = "0c4hxih9ljmv0d3izq5idyspk5zci26gdb6lv9klwcshwrfkvxj0";
   };
 
-  propagatedBuildInputs = [ pyyaml ]
+  propagatedBuildInputs = [ pyyaml requests ]
     ++ lib.optionals isPy27 [ configparser enum34 future functools32 ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/detect-secrets/default.nix:31]: They dropped python two support. Please remove all leftovers and disable it on python 2.7.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pyahocorasick/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyahocorasick/default.nix]: Updated.

File: pkgs/development/tools/detect-secrets/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/development/tools/detect-secrets/default.nix]: Removed

File: pkgs/development/tools/detect-secrets/default.nix (hunk 5 above)
Comment (fabaff) [reply pkgs/development/tools/detect-secrets/default.nix]: Disabled for Python 2.7.

File: pkgs/development/tools/detect-secrets/default.nix [hunk 6]
```diff
@@ -1,29 +1,52 @@
...
+, requests
+, responses
+, unidiff
+}:
 
 buildPythonApplication rec {
   pname = "detect-secrets";
-  version = "0.12.4";
+  version = "0.14.3";
+  disable = isPy27;
```
Comment (marsam) [inline pkgs/development/tools/detect-secrets/default.nix:21]:
```suggestion
  disabled = isPy27;
```

File: pkgs/development/tools/detect-secrets/default.nix [hunk 7]
```diff
@@ -1,29 +1,52 @@
...
 
-  # PyPI tarball doesn't ship tests
```
Comment (marsam) [inline pkgs/development/tools/detect-secrets/default.nix:7]: please, don't remove this comment, it refers about why we don't use the usual `fetchPypi` for fetching the source

File: pkgs/development/tools/detect-secrets/default.nix (hunk 6 above)
Comment (fabaff) [reply pkgs/development/tools/detect-secrets/default.nix:21]: Thanks, fixed.

File: pkgs/development/tools/detect-secrets/default.nix (hunk 7 above)
Comment (fabaff) [reply pkgs/development/tools/detect-secrets/default.nix:7]: Sure, we can keep it.

File: pkgs/development/tools/detect-secrets/default.nix [hunk 8]
```diff
@@ -1,29 +1,53 @@
-{ lib, buildPythonApplication, fetchFromGitHub, isPy27, pyyaml, unidiff, configparser, enum34, future, functools32, mock, pytest }:
+{ lib
+, buildPythonApplication
+, configparser
+, enum34
```
Comment (SuperSandro2000) [inline pkgs/development/tools/detect-secrets/default.nix:4]:
```suggestion
```

File: pkgs/development/tools/detect-secrets/default.nix [hunk 9]
```diff
@@ -1,29 +1,53 @@
...
+, fetchFromGitHub
+, functools32
+, future
+, isPy27
```
Comment (SuperSandro2000) [inline pkgs/development/tools/detect-secrets/default.nix:8]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107820` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>detect-secrets</li>
<li>python37Packages.pyahocorasick</li>
<li>python38Packages.pyahocorasick</li>
<li>python39Packages.pyahocorasick</li>
</ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 107820 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  <ul>    <li>detect-secrets</li>    <li>python37Packages.pyahocorasick</li>    <li>python38Packages.pyahocorasick</li>    <li>python39Packages.pyahocorasick</li>  </ul></details>

**Pull request merged**